### PR TITLE
[CAI-162] Chatbot/Save masked questions and answers using Presidio

### PR DESCRIPTION
**Pull request by batdevis**
#### List of Changes
Save masked questions and answers using Presidio.

#### Motivation and Context
Privacy.

#### How Has This Been Tested?
my local environment

### CODE REVIEW

File: apps/chatbot/src/modules/chatbot.py [hunk 1]
```diff
@@ -127,10 +127,13 @@ def _unmask_reference(self, response_str: str, nodes) -> str:
     
 
     def mask_pii(self, message: str) -> str:
-        try:
-            return self.pii.mask_pii(message)
-        except Exception as e:
-            logging.warning(f"[chatbot.py - mask_pii] exception in mask_pii: {e}")
+        if USE_PRESIDIO:
```
Comment (mdciri) [inline apps/chatbot/src/modules/chatbot.py:130]: is this env var still necessary or can we remove it?

File: apps/chatbot/src/modules/vector_database.py [hunk 2]
```diff
@@ -235,7 +235,7 @@ def build_automerging_index_redis(
     logging.info(f"[vector_database.py - build_automerging_index_redis] hash_table_{INDEX_ID} is now on Redis.")
 
     logging.info(f"[vector_database.py - build_automerging_index_redis] Creating index {NEW_INDEX_ID} ...")
-    nodes = Settings.node_parser.get_nodes_from_documents(documents)
+    nodes = Settings.node_parser.get_nodes_from_documents(documents[:2])
```
Comment (mdciri) [inline apps/chatbot/src/modules/vector_database.py:238]: remove [:2]

**Review by mdciri — CHANGES_REQUESTED** [review]

File: apps/chatbot/src/modules/chatbot.py (hunk 1 above)
Comment (batdevis) [reply apps/chatbot/src/modules/chatbot.py:130]: for now we keep it

File: apps/chatbot/src/modules/vector_database.py (hunk 2 above)
Comment (batdevis) [reply apps/chatbot/src/modules/vector_database.py]: done

**Review by mdciri — APPROVED** [review]

**Pull request merged**